fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorEventProcessor.java

```diff
@@ -499,6 +499,12 @@ private void processCreatePartition(CreatePartitionEvent createPartitionEvent) {
     }
 
     private void processDropTable(DropTableEvent dropTableEvent) {
+        // When drop a table, drop its snapshot store meanwhile.
```

Comment:
Only when the table is a primary key table, do we need to remove snapshot store
```
// When drop a primary key table, drop its snapshot store meanwhile.
        if (coordinatorContext.getTableInfo(dropTableEvent.getTableId()).hasPrimaryKey()) {
            Set<TableBucket> deleteTableBuckets =
                    coordinatorContext.getAllBucketsForTable(dropTableEvent.getTableId());
            completedSnapshotStoreManager.onRemoveCompletedSnapshotStoreByTableBuckets(
                    deleteTableBuckets);
        }
```

You'll need to add a method `getTableInfo` in `CoordinatorContext`